mantidproject/mantid
# Parallelize more of CI pipeline

**Description of work.**

**mantidproject/conda-recipes#77 must be merged at the same time**

Creating a new mantiddocs package, mantidproject/conda-recipes#77, required that the pipeline be rearranged. This has the advantage of parallelizing more of it. This work has aimed to reduce the duplication of the original by introducing a matrix over platforms and hiding details in the implementation functions.



**To test:**

See full completed [test job](https://builds.mantidproject.org/job/build_packages_from_branch_parallel/27/)



*This does not require release notes* because **it is an internal developer change**



---

#### Reviewer ####

Please comment on the following ([full description](http://developer.mantidproject.org/ReviewingAPullRequest.html)):

##### Code Review #####

- Is the code of an acceptable quality?
- Does the code conform to the [coding standards](http://developer.mantidproject.org/Standards/)?
- Are the unit tests small and test the class in isolation?
- If there is GUI work does it follow the [GUI standards](http://developer.mantidproject.org/Standards/GUIStandards.html)?
- If there are changes in the release notes then do they describe the changes appropriately?
- Are the release notes saved in a separate file, using Issue or PR number for file name and in the correct location?

##### Functional Tests #####

- Do changes function as described? Add comments below that describe the tests performed?
- Do the changes handle unexpected situations, e.g. bad input?
- Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.
